tests/test_install.py

```diff
@@ -207,3 +213,150 @@ def test_read_no_link(tmpdir):
     _make_lines_file(no_softlink)
     s2 = read_no_link(tempdir)
     assert s2 == {"line 1", "line 2", "line 4"}
+
+
+def test_install_freezes_env_by_default(
+    test_recipes_channel: Path,
+    tmp_env: TmpEnvFixture,
+    conda_cli: CondaCLIFixture,
+):
+    with tmp_env("dependent=2.0") as prefix:
+        assert package_is_installed(prefix, "dependent=2.0")
+        # Install a version older than the last one
+        conda_cli("install", f"--prefix={prefix}", "dependent=1.0", "--yes")
+
+        stdout, stderr, _ = conda_cli("list", f"--prefix={prefix}", "--json")
+
+        pkgs = json_loads(stdout)
+
+        conda_cli(
+            "install",
+            f"--prefix={prefix}",
+            "another_dependent",
+            "--freeze-installed",
+            "--yes",
+        )
+
+        stdout, _, _ = conda_cli("list", f"--prefix={prefix}", "--json")
+        pkgs_after_install = json_loads(stdout)
+
+        for pkg in pkgs:
+            for pkg_after in pkgs_after_install:
+                if pkg["name"] == pkg_after["name"]:
+                    assert pkg["version"] == pkg_after["version"]
+
+
+def test_install_mkdir(tmp_env: TmpEnvFixture, conda_cli: CondaCLIFixture):
+    try:
+        with tmp_env() as prefix:
+            file = prefix / "tempfile.txt"
+            file.write_text("test")
+            dir = prefix / "conda-meta"
+            assert isdir(dir)
+            assert exists(file)
+            with pytest.raises(DirectoryNotACondaEnvironmentError):
+                stdout, _, _ = conda_cli(
+                    "install", f"--prefix={dir}", "python", "--mkdir", "--yes"
+                )
+                assert (
+                    "The target directory exists, but it is not a conda environment."
+                    in stdout
+                )
+
+            conda_cli("create", f"--prefix={dir}", "--yes")
+            conda_cli("install", f"--prefix={dir}", "python", "--mkdir", "--yes")
+            assert package_is_installed(dir, "python")
+
+            rm_rf(prefix, clean_empty_parents=True)
+            assert path_is_clean(dir)
+
+            # regression test for #4849
+            conda_cli(
+                "install",
+                f"--prefix={dir}",
+                "python-dateutil",
+                "python",
+                "--mkdir",
+                "--yes",
+            )
+            assert package_is_installed(dir, "python")
+            assert package_is_installed(dir, "python-dateutil")
+
+    finally:
+        rm_rf(prefix, clean_empty_parents=True)
```

Comment:
a try-finally with `rm_rf(prefix)` is no longer required since we use `tmp_env` which uses our `path_factory` fixture which in turn uses the `tmp_path` fixture which automatically cleans up the temporary directory upon test completion